File: 41 First Missing Positive/41first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        i = 0

        while i < n and nums[i] < 1:
            i += 1

        c = 1
        while True:
            if i < n and nums[i] == c:
                while i < n and nums[i] == c:
                    i +=1
            else:
                return c
            c += 1
        return c

File: 41 First Missing Positive/41first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int i = 0, c = 1, n = nums.size();
        while(i < n && nums[i] < 1){i++;}

        while(true){
            if (i < n && nums[i] == c){
                while(i < n && nums[i] == c){i++;}
            } else {return c;}
            c++;
        }
        return c;
    }
};